sympy/physics/continuum_mechanics/beam.py

```diff
@@ -470,6 +502,12 @@ def apply_support(self, loc, type="fixed"):
         + 10*SingularityFunction(x, 20, -1)
         """
         loc = sympify(loc)
+
+        if loc in self._applied_rotation_hinges and type == "fixed":
```

Comment:
If loc is a floating point value, it is possible `loc in [list of floats]` will not find it due to ineact matches on floats.